libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -522,8 +524,34 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
                 return true;
             }
 
-			boolean isDone = uri.toString().replace("///", "/").toLowerCase(Locale.US).startsWith(loginOptions.getOauthCallbackUrl().replace("///", "/").toLowerCase(Locale.US));
-            if (isDone) {
+            // Check if user entered a custom domain
+            if (SalesforceSDKManager.getInstance().shouldInferCustomDomain() && isNewLoginUrl(uri)) {
+                try {
+                    String baseUrl = "https://" + uri.getHost();
+                    LoginServerManager serverManager = SalesforceSDKManager.getInstance().getLoginServerManager();
+                    LoginServerManager.LoginServer loginServer = serverManager.getLoginServerFromURL(baseUrl);
+
+                    // Check if url is already in server list
+                    if (loginServer == null) {
+                        // Add also sets as selected
+                        serverManager.addCustomLoginServer("Custom Domain", baseUrl);
+                    } else {
+                        serverManager.setSelectedLoginServer(loginServer);
+                    }
+
+                    // Set title to new login url
+                    loginOptions.setLoginUrl(baseUrl);
+                    // Checks the config for the selected login server
+                    (new AuthConfigTask(this)).execute();
+                } catch (Exception e) {
+                    SalesforceSDKLogger.e(TAG, "Unable to retrieve auth config.");
+                }
+            }
+
+            String formattedUrl = uri.toString().replace("///", "/").toLowerCase(Locale.US);
+            String callbackUrl = loginOptions.getOauthCallbackUrl().replace("///", "/").toLowerCase(Locale.US);
+			boolean authFlowFinished = formattedUrl.startsWith(callbackUrl);
```

Comment:
NIT what's up with indentation here?

Reply:
Github not interpreting a tab as 4 spaces?  Looks fine in the IDE but I'll fix it in my next commit.